### ISSUE TITLE
Projection to tangent space in Stiefel and Grassmannian fails on einsum

### ISSUE THREAD

**Issue by ninamiolane** [geomstats/geomstats#1020]
One submission of the ICLR 2021 challenge reports the following bug:

"We were unable to project points from the Grassmannian or the Stiefel manifold to the tangent space using Geomstats combined with the default canonical metric, or our custom metrics. We tried both `geomstats.learning.preprocessing.ToTangentSpace` and the newer `geomstats.geometry.grassmannian.Grassmannian.to_tangent`, but both are failing with the same einsum issue."

**Comment by nguigs** [comment]
The two methods are different:
- `Grassmannian.to_tangent` projects a matrix (`shape=[..., n,n]`) to the tangent space at a given base point (`shape=[n,n]`or must match that of the matrix). The following example works fine on the master branch:
```python
import geomstats.backend as gs
from geomstats.geometry.grassmannian import Grassmannian

n, k = 3, 2
n_samples = 4
space = Grassmannian(3, 2)
base_point = space.random_point()
vec = gs.random.rand(n_samples, n, n)
tangent_vec = space.to_tangent(vec, base_point)
print(space.is_tangent(tangent_vec, base_point))
```
Similarly on the Stiefel manifold
```python
import geomstats.backend as gs
from geomstats.geometry.stiefel import Stiefel

n, k = 3, 2
n_samples = 4
space = Stiefel(3, 2)
base_point = space.random_point()
vec = gs.random.rand(n_samples, n, k)
tangent_vec = space.to_tangent(vec, base_point)
print(space.is_tangent(tangent_vec, base_point))
```

- `ToTangentSpace.fit` is quite different and explained in this [notebook](https://github.com/geomstats/geomstats/blob/master/notebooks/03_simple_machine_learning_on_tangent_spaces.ipynb). It computes the Riemannian logarithms of input points at a point which is either given or computed as the Frechet mean of the input data. In a space represented by matrices (as is the case here), the logs are reshaped to 1d vectors so that usual ml algorithm can be applied. For the Grassmann manifold, tangent vectors are symmetric matrices, so the output of `ToTangentSpace.fit_transform` has `shape=[n_samples, n(n+1) / 2 ]`
The class `ToTangentSpace` behaves like a scikit-learn transformer. Here's an example on the Grassmannian:
```python
import geomstats.backend as gs
from geomstats.geometry.grassmannian import Grassmannian
from geomstats.geometry.symmetric_matrices import SymmetricMatrices
from geomstats.learning.preprocessing import ToTangentSpace

n, k = 3, 2
n_samples = 4
space = Grassmannian(3, 2)

transformer = ToTangentSpace(space)
data = space.random_uniform(n_samples) # shape= (n_samples, n, n)
logs = transformer.fit_transform(data) # shape= (n_samples, n(n+1)/2)
inverse = transformer.inverse_transform(logs) # recover original data
print(gs.allclose(inverse, data, atol=1e-8))

base_point = transformer.estimator.estimate_
tangent_vecs = SymmetricMatrices.from_vector(logs) # reshape vector to symmetric matrix
print(space.is_tangent(tangent_vecs, base_point, atol=1e-7)) # check that it is indeed a tangent vector
```
There indeed seem to need a few fixes to use this transformer on the Stiefel manifold
